src/Aframe/arcball-controller.js

```diff
@@ -90,24 +90,24 @@ THREE.TrackballControls = function (object, domElement) {
 
   // events
 
-  var changeEvent = { type: "change" };
-  var startEvent = { type: "start" };
-  var endEvent = { type: "end" };
+  let changeEvent = { type: "change" };
+  let startEvent = { type: "start" };
+  let endEvent = { type: "end" };
 
   // methods
 
   this.handleResize = function () {
-    var box = this.domElement.getBoundingClientRect();
+    let box = this.domElement.getBoundingClientRect();
     // adjustments come from similar code in the jquery offset() function
-    var d = this.domElement.ownerDocument.documentElement;
+    let d = this.domElement.ownerDocument.documentElement;
     this.screen.left = box.left + window.pageXOffset - d.clientLeft;
     this.screen.top = box.top + window.pageYOffset - d.clientTop;
     this.screen.width = box.width;
     this.screen.height = box.height;
   };
 
-  var getMouseOnScreen = (function () {
-    var vector = new THREE.Vector2();
+  let getMouseOnScreen = (function () {
```

Comment:
This is the self-invoking function I'd move away with. Cool use of closures, but hard to read

Reply:
This is the point where I start considering making classes instead of bags of functions and closures. Maybe a utility class for dealing with mouse position?